pkg/config/validation/validation.go

```diff
@@ -1514,21 +1523,55 @@ func ValidateAuthorizationPolicy(_, _ string, msg proto.Message) error {
 }
 
 // ValidateRequestAuthentication checks that request authentication spec is well-formed.
-func ValidateRequestAuthentication(_, _ string, msg proto.Message) error {
+func ValidateRequestAuthentication(name, namespace string, msg proto.Message) error {
 	in, ok := msg.(*security_beta.RequestAuthentication)
 	if !ok {
 		return errors.New("cannot cast to RequestAuthentication")
 	}
-	// TODO(diemtvu) add more details validation.
+
 	var errs error
+	errs = appendErrors(errs, validateWorkloadSelector(in.Selector))
+
 	for _, rule := range in.JwtRules {
-		if len(rule.Issuer) == 0 {
-			errs = appendErrors(errs, fmt.Errorf("issuer must be set"))
-		}
+		errs = appendErrors(errs, validateJwtRule(rule))
 	}
 	return errs
 }
 
+func validateJwtRule(rule *security_beta.JWT) (errs error) {
```

Comment:
are the same code reused by both analyzer and validation webhook? @sushicw @selmanj

Reply:
Yes, this should get invoked for both. You should be able to `make istioctl` and then run `istioctl x analyze <your_file_with_an_error.yaml>` and see it show an error.